segmentation/utils/spatial_analysis_test.py

```diff
@@ -52,84 +81,232 @@ def make_expression_matrix(enrichment_type):
+        all_data_pos.iloc[112:116, 31] = 1
+        all_data_pos.iloc[112:116, 32] = "Pheno1"
         # Assign column names to columns not for markers (columns to be excluded)
-        all_patient_data_pos = all_patient_data_pos.rename(excluded_colnames, axis=1)
+        all_patient_data_pos = all_data_pos.rename(excluded_colnames, axis=1)
         return all_patient_data_pos
     elif enrichment_type == "negative":
-        all_patient_data_neg = pd.DataFrame(np.zeros((60, 31)))
+        all_data_neg = pd.DataFrame(np.zeros((120, 33)))
         # Assigning values to the patient label and cell label columns
-        all_patient_data_neg[30] = "Point8"
-        all_patient_data_neg[24] = np.arange(len(all_patient_data_neg[1])) + 1
+        all_data_neg.loc[0:59, 30] = "Point8"
+        all_data_neg.loc[60:, 30] = "Point9"
+        all_data_neg.loc[0:59, 24] = np.arange(60) + 1
+        all_data_neg.loc[60:, 24] = np.arange(60) + 1
         # We create two groups of 20 cells positive for marker 1 (in column index 2)
         # and marker 2 (in column index 3) respectively.
         # The two populations are not within the dist_lim in dist_mat_neg (negative enrichment distance matrix)
-        all_patient_data_neg.iloc[0:20, 2] = 1
-        all_patient_data_neg.iloc[20:40, 3] = 1
+        all_data_neg.iloc[0:20, 2] = 1
+        all_data_neg.iloc[20:40, 3] = 1
+
+        all_data_neg.iloc[60:80, 3] = 1
+        all_data_neg.iloc[80:100, 2] = 1
+        # We assign the two populations of cells different cell phenotypes
+        all_data_neg.iloc[0:20, 31] = 1
+        all_data_neg.iloc[0:20, 32] = "Pheno1"
+        all_data_neg.iloc[60:80, 31] = 2
+        all_data_neg.iloc[60:80, 32] = "Pheno2"
+
+        all_data_neg.iloc[20:40, 31] = 2
+        all_data_neg.iloc[20:40, 32] = "Pheno2"
+        all_data_neg.iloc[80:100, 31] = 1
+        all_data_neg.iloc[80:100, 32] = "Pheno1"
         # Assign column names to columns not for markers (columns to be excluded)
-        all_patient_data_neg = all_patient_data_neg.rename(excluded_colnames, axis=1)
+        all_patient_data_neg = all_data_neg.rename(excluded_colnames, axis=1)
         return all_patient_data_neg
 
 
 def test_calculate_channel_spatial_enrichment():
+
+    excluded_colnames = ["cell_size", "Background", "HH3",
+                         "summed_channel", "cellLabelInImage", "area",
+                         "eccentricity", "major_axis_length", "minor_axis_length",
+                         "perimeter", "SampleID", "FlowSOM_ID", "cell_type"]
+
     # Test z and p values
 
     marker_thresholds = make_threshold_mat()
 
     # Positive enrichment
-    all_patient_data_pos = make_expression_matrix("positive")
-    dist_mat_pos = make_distance_matrix("positive")
+    all_data_pos = make_expression_matrix("positive")
+    # dist_mat_pos = make_distance_matrix("positive")
+    make_distance_matrix("positive")
+    dist_mat_pos = h5py.File("posmat4.hdf5", "r")
```

Comment:
Same thing, this is going to result in files being saved during testing, which we don't want. This function should return the dist mat and never save it, since we don't need to keep this file around after testing